scripts/ingest_startgg.py

```diff
@@ -3,190 +3,196 @@
-    p1_safe_name = normalize(player1)
-    p2_safe_name = normalize(player2)
-    match_url = None
-    url_indices_to_remove = []
-    for video_index, video_data in enumerate(playlist_urls):
-        if video_data is None:
-            continue
-        video_title, video_url = video_data
-        video_title_safe = normalize(video_title)
-        if (p1_safe_name in video_title_safe) and (p2_safe_name in video_title_safe):
-            # Prompts for input
-            print(f"Found potential match for '{player1}' vs '{player2}' ({round_text}): {video_title}")
-            prompt = input("Use? (y/n/remove): ")
-            if prompt.lower().startswith("y"):
-                match_url = video_url
-                url_indices_to_remove.append(video_index)
-                break
-            elif prompt.lower().startswith("r"):
-                url_indices_to_remove.append(video_index)
-    for url_index in sorted(url_indices_to_remove, reverse=True):
-        del playlist_urls[url_index]
-    return match_url
-
-def process_event(startgg_url, playlist_urls):
-    print(f"Processing {startgg_url}")
+def get_tournament_name_sets_and_timezone(slug: str):
+    tournament_response = requests.get(f"https://api.start.gg/tournament/{slug}?expand[]=groups&expand[]=phase").json()
+    name = tournament_response["entities"]["tournament"]["name"]
+    time = tournament_response["entities"]["tournament"]["startAt"]
+    timezone = tournament_response["entities"]["tournament"]["timezone"]
+    date = datetime.datetime.fromtimestamp(time, tz=ZoneInfo(timezone)).strftime("%Y-%m-%d")
+    phase_id_to_name = dict[str, str]()
+    for phase in tournament_response["entities"]["phase"]:
+        phase_id_to_name[phase["id"]] = phase["name"]
+    sets = []
+    for group in tournament_response["entities"]["groups"]:
+        group_response = requests.get(f"https://api.start.gg/phase_group/{group["id"]}?expand[]=sets&expand[]=entrants").json()
+        entrant_id_to_gamer_tags = dict[int, list[str]]()
+        for entrant in group_response["entities"]["entrants"]:
+            gamerTags: list[str] = []
+            for key in entrant["mutations"]["participants"]:
+                gamerTags.append(entrant["mutations"]["participants"][key]["gamerTag"])
+            entrant_id_to_gamer_tags[entrant["id"]] = gamerTags
+        for set in group_response["entities"]["sets"]:
+            if (set["entrant1Id"] is not None and set["entrant2Id"] is not None):
+                sets.append({
+                    "id": set["id"],
+                    "full_round_text": set["fullRoundText"],
+                    "phase_name": phase_id_to_name[set["phaseId"]],
+                    "entrant_1_gamer_tags": entrant_id_to_gamer_tags[set["entrant1Id"]],
+                    "entrant_2_gamer_tags": entrant_id_to_gamer_tags[set["entrant2Id"]],
+                    "entrant_1_character_ids": set["entrant1CharacterIds"] if "entrant1CharacterIds" in set else [],
```

Comment:
`Dict.get` allows you to set a fallback value

```suggestion
                    "entrant_1_character_ids": set.get("entrant1CharacterIds", [])
```